lightning/common/util_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common_test
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/go-sql-driver/mysql"
+	. "github.com/pingcap/check"
+	tmysql "github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb-lightning/lightning/common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type utilSuite struct{}
+
+func TestUtil(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&utilSuite{})
+
+func (s *utilSuite) TestDirNotExist(c *C) {
+	c.Assert(common.IsDirExists("."), IsTrue)
+	c.Assert(common.IsDirExists("not-exists"), IsFalse)
+}
+
+func (s *utilSuite) TestGetJSON(c *C) {
+	type TestPayload struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}
+	var request = TestPayload{
+		Username: "lightning",
+		Password: "lightning-ctl",
+	}
+
+	// Mock success response
+	handle := func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(http.StatusOK)
+		err := json.NewEncoder(res).Encode(request)
+		c.Assert(err, IsNil)
+	}
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		handle(res, req)
+	}))
+	defer testServer.Close()
+	response := TestPayload{}
+	err := common.GetJSON(http.DefaultClient, "http://not-exists", &response)
+	c.Assert(err, NotNil)
+	err = common.GetJSON(http.DefaultClient, testServer.URL, &response)
+	c.Assert(err, IsNil)
+	c.Assert(request, DeepEquals, response)
+
+	// Mock `StatusNoContent` response
+	handle = func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(http.StatusNoContent)
+	}
+	err = common.GetJSON(http.DefaultClient, testServer.URL, &response)
+	c.Assert(err, NotNil)
+	c.Assert(strings.Contains(err.Error(), "http status code != 200"), IsTrue)
```

Comment:
Combine these two lines (79–80) into:

```suggestion
	c.Assert(err, ErrorMatches ".*http status code != 200.*")
```

Reply:
@lonng I mean `ErrorMatches` already implies `NotNil`. https://github.com/pingcap/check/blob/67f458068fc864dabf17e38d4d337f28430d13ed/checkers.go#L260-L262